EnhancedMediaPlayer/AVPlayerView.swift

```diff
@@ -0,0 +1,71 @@
+// AVPlayerView.swift
+
+import AVKit
+import SwiftUI
+
+public struct AVPlayerView: UIViewControllerRepresentable {
+    private var playerState: MediaPlayerView.PlayerState
+    private let mediaURL: URL
+    private let onMediaFinished: () -> Void
+
+    init(
+        mediaURL: URL,
+        playerState: MediaPlayerView.PlayerState,
+        onMediaFinished: @escaping () -> Void
+    ) {
+        self.mediaURL = mediaURL
+        self.playerState = playerState
+        self.onMediaFinished = onMediaFinished
+    }
+
+    public func makeUIViewController(context: Context) -> AVPlayerViewController {
+            let playerController = AVPlayerViewController()
+            setupPlayerController(playerController)
+            updatePlayerState(playerController)
+            return playerController
+        }
+
+    public func updateUIViewController(_ playerController: AVPlayerViewController, context: Context) {
+        if let currentURL = playerController.player?.currentItem?.asset as? AVURLAsset,
+           currentURL.url != mediaURL {
+            setupPlayerController(playerController)
+        }
+        updatePlayerState(playerController)
+    }
+
+    private func setupPlayerController(_ playerController: AVPlayerViewController) {
+        playerController.showsPlaybackControls = false
+        playerController.player = AVPlayer(url: mediaURL)
+        playerController.observeMediaFinished {
+            onMediaFinished()
+        }
+    }
+    
+    private func updatePlayerState(_ playerController: AVPlayerViewController) {
+        switch playerState {
+            case .playing: playerController.player?.play()
+            default: playerController.player?.pause()
+        }
+    }
+}
+
+extension AVPlayerViewController {
+    func observeMediaFinished(onEnd: @escaping () -> Void) {
```

Comment:
`private`

Reply:
if I make it `private` I won't be able to access it from outside of `AVPlayerViewController `